Editor/CourseController/UI/CourseControllerSetupEditor.cs

```diff
@@ -51,6 +52,13 @@ private void OnEnable()
             }
             
             currentRequiredComponents = availableCourseControllers[selectedIndex].GetRequiredSetupComponents();
+            currentRequiredComponents.AddRange(currentRequiredComponents
+                .SelectMany(type => type.GetCustomAttributes(typeof(RequireComponent)).Cast<RequireComponent>())
+                .SelectMany(component => new List<Type>() {component.m_Type0, component.m_Type1, component.m_Type2})
+                .Where(type => type != null)
+                .Distinct()
+                .Except(currentRequiredComponents)
+                .ToList());
```

Comment:
Wow, I almost had a stroke trying to read that 🤕
What does this do? What does it for? What is the meaning of life?

Reply:
damn I failed

Reply:
it also adds all components required via RequiredComponent() to the currentRequiredComponents list.

Reply:
![image](https://user-images.githubusercontent.com/6911992/116705123-81246280-a9cc-11eb-9291-652559f3f7d8.jpeg)

Reply:
I think you did the same somewhere in editor, but i did not find the method :D

Reply:
Na it's fine 😅